### PR TITLE
Show_discounts_Pull1[P0]

### PR DESCRIPTION
**Pull request by khushi2305**
This page displays all the discounts offered by the company . 
Important features -
1) Each discount offer is either on a single device or a combination of devices .The discount corresponding to each offer is displayed on the left 
2) The discount offer list is made  scrollable and the offer gets selected by clicking on the container . On clicking the color of the container changes .
3) Offers can be selected / deselected any number of times. Selected discount offers are added to the list and unselected ones are removed .
4) The page has the 'Get Discounts" tab which will be used to launch the resource picker .

Show Policies 1
Show Policies 2

### CODE REVIEW

File: home_insurance_app/lib/pages/show_discounts.dart [hunk 1]
```diff
@@ -8,6 +14,181 @@ class DisplayDiscounts extends StatefulWidget {
...
+                    label: Text(
+                      'Get Discounts',
+                      style: TextStyle(
+                        color: Colors.black87,
+                        fontSize: 15.0,
+                        fontWeight: FontWeight.w900,
+                        fontFamily: "PTSerifBI",
+                      ),
+                    ),
+                    onPressed: () {}, // resource picker url is launched
```
Comment (dreamplay) [inline home_insurance_app/lib/pages/show_discounts.dart:69]: let's add this url.
also how do we redirect the user back to flow from resource picker?

File: home_insurance_app/lib/pages/show_discounts.dart [hunk 2]
```diff
@@ -1,4 +1,10 @@
 import 'package:flutter/material.dart';
+import 'package:homeinsuranceapp/data/company_offers.dart';
+import 'package:homeinsuranceapp/data/offer.dart';
+
+Map data;
+//Offers selected by the user
+List<Offer> selectedOffers = new List<Offer>();
```
[the remark this answers is not in the document]
Comment (khushi2305) [reply home_insurance_app/lib/pages/show_discounts.dart:7]: No , more than 1 offers can be selected . Selected offers are then stored in a list .

File: home_insurance_app/lib/data/offer.dart [hunk 3]
```diff
@@ -0,0 +1,10 @@
+//This class is a structure for each discount offer given by the company
+class Offer {
+  Map<String, int>
+      requirements; // This map stores each device type and their corresponding number required for this offer
```
[the remark this answers is not in the document]
Comment (khushi2305) [reply home_insurance_app/lib/data/offer.dart:4]: Done

File: home_insurance_app/lib/pages/show_discounts.dart (hunk 1 above)
Comment (khushi2305) [reply home_insurance_app/lib/pages/show_discounts.dart:70]: Removed external fonts . Added Inbuilt font . We have to look over this that how can we redirect back from resource picker .

**Comment by anjalisoni3655** [comment]
@googlebot I consent.

File: home_insurance_app/lib/pages/show_discounts.dart [hunk 4]
```diff
@@ -8,6 +14,182 @@ class DisplayDiscounts extends StatefulWidget {
...
+                              children: (CompanyOffers.availableOffers[index])
+                                  .requirements
+                                  .entries
+                                  .map(
+                                    (entry) => Align(
+                                      alignment: Alignment.topLeft,
+                                      child: Container(
+//                                          margin: EdgeInsets.symmetric(
+//                                              vertical: screenheight / 100,
+//                                              horizontal: screenwidth / 100),
```
Comment (akashagarwal) [inline home_insurance_app/lib/pages/show_discounts.dart:157]: nit : no commented code.

File: home_insurance_app/lib/pages/show_discounts.dart [hunk 5]
```diff
@@ -8,6 +14,182 @@ class DisplayDiscounts extends StatefulWidget {
...
+                  fontFamily: 'Source Sans Pro',
+                ),
+              ),
+            ),
+            Divider(
+              color: Colors.brown,
+              height: screenheight / 100,
+              thickness: screenheight / 100,
+              indent: screenwidth / 50,
+              endIndent: screenwidth / 50,
```
Comment (akashagarwal) [inline home_insurance_app/lib/pages/show_discounts.dart:47]: can we use css files for formatting ?
that way we can easily apply consistent formatting across all pages of the app without duplicating this information

File: home_insurance_app/lib/pages/show_discounts.dart (hunk 4 above)
Comment (khushi2305) [reply home_insurance_app/lib/pages/show_discounts.dart]: Removed

File: home_insurance_app/lib/pages/show_discounts.dart (hunk 5 above)
Comment (khushi2305) [reply home_insurance_app/lib/pages/show_discounts.dart]: Done . Please check the way I have done is correct or not .

File: sdk/.gitignore [hunk 6]
```diff
@@ -59,7 +59,7 @@ build/
 **/ios/Flutter/App.framework
 **/ios/Flutter/Flutter.framework
 **/ios/Flutter/Flutter.podspec
-**/ios/Flutter/Generated.xcconfig
+**/ios/Flutter/Generated.xcconfighoe
```
Comment (dreamplay) [inline sdk/.gitignore:62]: xcconfig was wrong?

**Comment by oshhh** [comment]
@googlebot I consent.

**Pull request merged**